internal/controller/controllers/agent_controller_test.go

```diff
@@ -445,6 +449,43 @@ var _ = Describe("agent reconcile", func() {
 		Expect(conditionsv1.FindStatusCondition(agent.Status.Conditions, v1beta1.SpecSyncedCondition).Status).To(Equal(corev1.ConditionTrue))
 	})
 
+	It("Agent bind, cluster not found in DB", func() {
```

Comment:
I would also add a test that after a retry cluster is found in DB, so we can be sure it recovers as we expected it to and update conditions accordingly.

Reply:
Done

Reply:
👍